src/database.cpp

```diff
@@ -156,7 +183,17 @@ std::string Database::open(bool wait, bool memory) {
   while (true) {
     char *fail;
     ret = sqlite3_exec(imp->db, schema_sql, 0, 0, &fail);
-    if (ret == SQLITE_OK) break;
+    if (ret == SQLITE_OK) {
+      const char *get_version = "select (select count(row_id) from entropy), (select max(version) from schema);";
```

Comment:
What is this `entropy` thing about?

Reply:
Wake uses hashes to create signatures for some functions in build rules. If those build rules change, they get re-executed. The hash function I use needs a seed to prevent intentional collision attacks. The first time the database is created, a persistent random seed is chosen which will be used for all hashes. That way the hashes in the database do not change values between runs.

In this context, I am using the entropy as a means to detect that the database is "fresh"/unused. That's needed because earlier versions do not have a schema(version) to check.

Reply:
OK, thanks for the additional context. I think it's worth adding a comment to describe that the choice of table here is not really important, and that it's just to detect a fresh DB. As an outside reader of the code, I was reading more into the entropy than was actually relevant for this specific use case.

Reply:
Comment added.

Reply:
Thanks!